src/scripts/main.js

```diff
@@ -1 +1,68 @@
 'use strict';
+
+function firstPromise() {
+  return new Promise((resolve, reject) => {
+    document.body.addEventListener('click', () => {
+      resolve('First promise was resolved');
+    });
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  });
+}
+
+function secondPromise() {
+  return new Promise((resolve) => {
+    document.body.addEventListener('click', () => {
+      resolve('Second promise was resolved');
+    });
+
+    document.body.addEventListener('contextmenu', (e) => {
+      e.preventDefault();
+      resolve('Second promise was resolved');
+    });
```

Comment:
The 'secondPromise' function resolves on both 'click' and 'contextmenu' events. This means if a user triggers both events, the promise will resolve twice, which is not ideal for a promise. Consider adding a flag to ensure the promise resolves only once.